index.html

```diff
@@ -0,0 +1,349 @@
+        }
+      }
+
+      class Cache {
+        constructor(key) {
+          this.key = key;
+        }
+
+        get() {
+          const cache = JSON.parse(localStorage.getItem(this.key));
+          if (cache?.expirationTime > Date.now()) {
+            return cache;
+          }
+
+          return null;
+        }
+
+        set(data, expiration = 60 * 60 * 1000) {
+          localStorage.setItem(
+            this.key,
+            JSON.stringify({
+              data,
+              expirationTime: Date.now() + expiration,
+            })
+          );
+        }
+      }
+
+      async function loadProjects({
+        container,
+        sort,
+        limit = 3,
+        metaFields,
+        loader,
+      }) {
+        const url = `https://api.github.com/search/repositories?q=user:MauricioRobayo&sort=${sort}&per_page=${limit}`;
+        const header = createHeader(url, loader);
+        container.append(header);
+
+        loader.start();
+
+        const data = await fetchRepos({
+          url,
+          cache: new Cache(`${sort}-${limit}`),
+        });
+
+        if (data.error) {
+          loader.stop("✗");
+
+          container.classList.add("error");
+          container.append(data.message);
+        } else {
+          loader.stop("✔");
+
+          if (data.cacheHit) {
+            const cacheStatus = createCacheStatus(data.expirationTime);
+            header.append(cacheStatus);
+          } else {
+            const rateLimit = createRateLimit(data.rateLimit);
+            header.append(...rateLimit);
+          }
+
+          const body = createBody();
+          const repos = createRepos({
+            container: body,
+            repos: data.repos,
+            metaFields,
+          });
+          body.append(...repos);
+          container.append(body);
+        }
+      }
+
+      function createRateLimit(rateLimit) {
+        return rateLimit.map((info) => {
+          const p = document.createElement("p");
+          p.textContent = info.join(": ");
+          return p;
+        });
+      }
+
+      function createHeader(url, loader) {
+        const header = document.createElement("header");
+        const title = document.createElement("h2");
+        const link = document.createElement("a");
+        link.href = url;
+        link.textContent = url;
+        title.append(loader.ref, "Fetching ", link);
+        header.append(title);
+
+        return header;
+      }
+
+      function createBody() {
+        const projectsBody = document.createElement("div");
+        projectsBody.classList.add("projects-body");
+
+        return projectsBody;
+      }
+
+      function createCacheStatus(expirationTime) {
+        const container = document.createElement("div");
+        container.innerHTML = `
+          <p>Cache hit!</p>
+          <p>Expires: ${expirationTime}</p>
+        `;
+        return container;
+      }
+
+      function createRepos({ container, repos, metaFields }) {
```

Comment:
The parameter `container` is not used here. Is that expected, or a bug? 🤔 
```suggestion
      function createRepos({ repos, metaFields }) {
```